### PR TITLE
Fixes SG-Accel #27 add validateSgReaderDbConfig + validateSgAccelDbCo…

### PR DESCRIPTION
**Pull request by tleyden**
…nfig

Fixes https://github.com/couchbaselabs/sync-gateway-accel/issues/27

https://github.com/couchbaselabs/sync-gateway-accel/pull/67 should be merged with this

### CODE REVIEW

File: examples/sg-accel-index-reader.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
```
Comment (adamcfraser) [inline examples/sg-accel-index-reader.json]: I feel like these should be moved to the sg-accel examples folder.

This one could be renamed 'sync-gateway-accel-enabled.json'.

File: examples/sg-accel-index-writer.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{
```
Comment (adamcfraser) [inline examples/sg-accel-index-writer.json]: As above, could be moved to the sg-accel examples folder, as sg-accel.json.

File: rest/config.go [hunk 3]
```diff
@@ -299,6 +300,43 @@ func (dbConfig DbConfig) validate() error {
 
 }
 
+func (dbConfig *DbConfig) validateSgReaderDbConfig() error {
```
Comment (adamcfraser) [inline rest/config.go]: In general I think we should move away from the 'reader/writer' terminology.  It was more appropriate when we had a single binary that could run in either mode.  Now that the binaries are split, it's either a Sync Gateway config, or a Sync Gateway Accel config.

**Review by adamcfraser — CHANGES_REQUESTED** [review]
Functionally it all looks good, but a few comments about naming and location of files.

File: rest/config.go (hunk 3 above)
Comment (tleyden) [reply rest/config.go]: Agreed.  How about `validateSGConfig()` and `validateSGAccelConfig()`?

File: examples/sg-accel-index-writer.json (hunk 2 above)
Comment (tleyden) [reply examples/sg-accel-index-writer.json]: Agreed

File: examples/sg-accel-index-reader.json (hunk 1 above)
Comment (tleyden) [reply examples/sg-accel-index-reader.json]: Agreed

**Pull request merged**